### PR TITLE
491 beanie for metadata, dataset_metadata

### PR DESCRIPTION
**Pull request by longshuicy**
endpoints tested

GET http://localhost:8000/api/v2/metadata/definition?skip=0&limit=100

GET http://localhost:8000/api/v2/metadata/definition?skip=0&limit=100&name=abstract

POST http://localhost:8000/api/v2/metadata/definition
```
{"name":"abstract","description":"abstract","context":[{"abstract":"http://purl.org/dc/terms/abstract"}],"fields":[{"name":"abstract","list":false,"widgetType":"TextField","config":{"type":"str"},"required":true}]}
```

POST /{dataset_id}/metadata worked

GET http://localhost:8000/api/v2/datasets/64590e8c7891dca20ce576a2/metadata works

DELETE http://localhost:8000/api/v2/datasets/64590e8c7891dca20ce576a2/metadata
```
{"id":"645920cdd0ed048b3f31b2be","definition":"abstract"}
```

PATCH http://localhost:8000/api/v2/datasets/64590e8c7891dca20ce576a2/metadata
```
{"id":"645920cdd0ed048b3f31b2be","definition":"abstract","content":{"abstract":"bbbb"}}
```

PUT http://localhost:8000/api/v2/datasets/64590e8c7891dca20ce576a2/metadata
```
{
    "id": "64592afa7ef21772aa89486f",
    "context": null,
    "context_url": null,
    "definition": "abstract",
    "content": {
        "abstract": "lalallalalala"
    },
    "description": "abstract",
    "resource": {
        "collection": "datasets",
        "resource_id": "64590e8c7891dca20ce576a2",
        "version": null
    },
    "agent": {
        "id": "64592afa7ef21772aa894870",
        "creator": {
            "id": "64590e677891dca20ce57699",
            "email": "cwang138@illinois.edu",
            "first_name": "Chen",
            "last_name": "Wang"
        },
        "listener": null
    },
    "created": "2023-05-08T17:01:46.066000"
}
```

### CODE REVIEW

File: backend/app/models/metadata.py [hunk 1]
```diff
@@ -117,9 +111,19 @@ class MetadataDefinitionIn(MetadataDefinitionBase):
...
     creator: UserOut
 
+    class Settings:
+        name = "metadata.definitions"
+        indexes = [
+            [
+                ("name", pymongo.TEXT),
+                ("description", pymongo.TEXT),
+            ],
+        ]
```
Comment (longshuicy) [inline backend/app/models/metadata.py:125]: remove not needed

File: backend/app/models/metadata.py [hunk 2]
```diff
@@ -312,26 +319,23 @@ def deep_update(orig: dict, new: dict):
...
-            metadata.get("definition", None),
-            metadata.get("context_url", None),
-            metadata.get("context", []),
-        )
-        metadata["content"] = updated_content
-        db["metadata"].replace_one(
-            {"_id": metadata["_id"]}, MetadataDB(**metadata).to_mongo()
+            metadata.dict().get("definition", None),
+            metadata.dict().get("context_url", None),
+            metadata.dict().get("context", []),
```
Comment (longshuicy) [inline backend/app/models/metadata.py:333]: replace with metadata.definition

File: backend/app/routers/metadata.py [hunk 3]
```diff
@@ -96,17 +98,18 @@ async def update_metadata(
...
+        allow: bool = Depends(MetadataAuthorization("editor")),
 ):
     """Delete metadata by specific ID."""
     if (
-        md := await db["metadata"].find_one({"_id": PyObjectId(metadata_id)})
+            md := await MetadataDB.find_one(MetadataDB.id == PyObjectId(metadata_id))
     ) is not None:
         # TODO: Refactor this with permissions checks etc.
         await db["metadata"].delete_one({"_id": PyObjectId(metadata_id)})
+        await MetadataDB(id=PyObjectId(metadata_id)).delete()
```
Comment (longshuicy) [inline backend/app/routers/metadata.py:112]: i can just do md.delete()

**Review by max-zilla — APPROVED** [review]

**Review by tcnichol — APPROVED** [review]
Routes tested and works. Marking approved.

**Pull request merged**